datafusion/optimizer/src/push_down_filter.rs

```diff
@@ -689,19 +700,15 @@ impl OptimizerRule for PushDownFilter {
                 }
             }
             LogicalPlan::CrossJoin(CrossJoin { left, right, .. }) => {
-                let predicates = utils::split_conjunction_owned(utils::cnf_rewrite(
-                    filter.predicate.clone(),
-                ));
-
+                let predicates = utils::split_conjunction_owned(filter.predicate.clone());
                 push_down_all_join(predicates, &filter.input, left, right, vec![])?
             }
             LogicalPlan::TableScan(scan) => {
                 let mut new_scan_filters = scan.filters.clone();
                 let mut new_predicate = vec![];
 
-                let filter_predicates = utils::split_conjunction_owned(
-                    utils::cnf_rewrite(filter.predicate.clone()),
```

Comment:
I believe this is the only place that calls `cnf_rewrite` 

  https://github.com/search?q=repo%3Aapache%2Farrow-datafusion%20cnf_rewrite&type=code

so if we are going to stop calling it, I think we should also remove `cnf_rewrite` from the code